Automattic/wp-calypso
# AT: handle inconsistent sites response during transfer

During some point in AT procedure, the `me/sites` response starts giving `.wordpress.com` URL for a while.
Also, sites list is initiated with either WpcomSite or JetpackSite and that needs to change.
That PR corrects the response coming to SitesList so that it does not destroy all calypso

## Movie
https://cloudup.com/cTABEs09OUN

## Testing procedure
- apply patch
- transfer site
- observe that no JS breaks

CC @lamosty @gwwar
